src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/GroupCollection.cs

```diff
@@ -210,12 +211,12 @@ void IList.RemoveAt(int index)
             return new Enumerator(this);
         }
 
-        public bool TryGetValue(string key, out Group value)
+        public bool TryGetValue(string key, [NotNullWhen(true)] out Group value)
```

Comment:
value should be `Group?`.

Reply:
This implements `IReadOnlyDictionary<Group>.TryGetValue()`

Reply:
Hmmm, should it instead implemente `IReadOnlyDictionary<group?>`.

Reply:
I don't think so, this is guaranteed to produce non-null Group instances. @stephentoub thoughts?

Reply:
I think it should implement `IReadOnlyDictionary<Group>`, but this method should be declared as `[NotNullWhen(true)] Group? group`, and if the compiler complains about what it sees as a mismatch with the interface, suppress the warning with a pragma.  That's the signature we would use if it wasn't implementing an interface, so it's consistent, and even if the compiler has trouble seeing that it's effectively the same, it is, but it's much clearer to a consumer, in that the API takes a `Group?` (which is accurate) but then the compiler is able to see inside a branch for it returning true that it is not-null.

Reply:
Following @MadsTorgersen's recommendation, I changed to `[MaybeNullWhen(false)]` and removed warning suppression.

Reply:
Done.